File: Data Structures & Algorithms/generate-parentheses/submission-0.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        res = []

        def backtrack(cur, open_count, close_count):
            if len(cur) == 2 * n:
                res.append(cur)
                return

            if open_count < n:
                backtrack(cur + "(", open_count + 1, close_count)

            if close_count < open_count:
                backtrack(cur + ")", open_count, close_count + 1)

        backtrack("", 0, 0)
        return res

File: Data Structures & Algorithms/count-prefix-and-suffix-pairs-i/submission-0.cpp
class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {
        int res = 0;
        for (int i =0; i < words.size() -1; i++){
            for(int j = i+1; j < words.size(); j++){
                if (
                    words[j].substr(0, words[i].size()) == words[i] &&
                    words[j].substr(words[j].size() - words[i].size(), words[i].size()) == words[i]
                ){
                res++;
                }
            }
            
        }
        return res;
    }
};